### PR TITLE
Flink: Update the docs

### PR DESCRIPTION
**Pull request by Fokko**
The docs were still mentioning Flink 1.11, and was long overdue for an overhaul. Updated the links and also did some rewriting of the docs by removing `we`.

The Python quickstart became much simpler since the jars now ship with the package itself: https://pypi.org/project/apache-flink-libraries/1.16.1/

### CODE REVIEW

File: docs/flink-getting-started.md [hunk 1]
```diff
@@ -46,156 +46,136 @@ Apache Iceberg supports both [Apache Flink](https://flink.apache.org/)'s DataStr
 
 ## Preparation when using Flink SQL Client
 
-To create iceberg table in flink, we recommend to use [Flink SQL Client](https://ci.apache.org/projects/flink/flink-docs-stable/dev/table/sqlClient.html) because it's easier for users to understand the concepts.
+To create Iceberg table in Flink, recommended is to use [Flink SQL Client](https://ci.apache.org/projects/flink/flink-docs-stable/dev/table/sqlClient.html) as it's easier for users to understand the concepts.
```
Comment (stevenzwu) [inline docs/flink-getting-started.md:49]: `recommended` -> `recommended way` or `it is recommended to`

File: docs/flink-getting-started.md [hunk 2]
```diff
@@ -46,156 +46,136 @@ Apache Iceberg supports both [Apache Flink](https://flink.apache.org/)'s DataStr
...
 
-Step.1 Downloading the flink 1.11.x binary package from the apache flink [download page](https://flink.apache.org/downloads.html). We now use scala 2.12 to archive the apache iceberg-flink-runtime jar, so it's recommended to use flink 1.11 bundled with scala 2.12.
+Download the Flink 1.16.1 binary package from the Apache Flink [download page](https://flink.apache.org/downloads.html). Iceberg uses Scala 2.12 when compiling the Apache `iceberg-flink-runtime` jar, so it's recommended to use Flink 1.16 bundled with Scala 2.12.
```
Comment (stevenzwu) [inline docs/flink-getting-started.md:51]: let's remove the version from this sentence

File: docs/flink-getting-started.md [hunk 3]
```diff
@@ -344,11 +324,11 @@ CREATE TABLE `hive_catalog`.`default`.`sample` (
 );
 ```
 
-Table create commands support the most commonly used [flink create clauses](https://ci.apache.org/projects/flink/flink-docs-release-1.11/dev/table/sql/create.html#create-table) now, including: 
+Table create commands support the commonly used [Flink create clauses](https://nightlies.apache.org/flink/flink-docs-release-1.16/docs/dev/table/sql/create/) including:
```
Comment (stevenzwu) [inline docs/flink-getting-started.md:327]: what about using the doc link from master branch? then we don't need to worry about the Flink versions in the doc
https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/create/

File: docs/flink-getting-started.md [hunk 4]
```diff
@@ -687,11 +664,11 @@ FlinkSink.forRowData(input)
 env.execute("Test Iceberg DataStream");
 ```
 
-The iceberg API also allows users to write generic `DataStream<T>` to iceberg table, more example could be found in this [unit test](https://github.com/apache/iceberg/blob/master/flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/sink/TestFlinkIcebergSink.java).
+The iceberg API also allows users to write generic `DataStream<T>` to iceberg table, more example could be found in this [unit test](https://github.com/apache/iceberg/blob/master/flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/sink/TestFlinkIcebergSink.java).
 
 ### Overwrite data
 
-To overwrite the data in existing iceberg table dynamically, we could set the `overwrite` flag in FlinkSink builder.
+To overwrite the data in existing iceberg table dynamically, set the `overwrite` flag in FlinkSink builder.
```
Comment (stevenzwu) [inline docs/flink-getting-started.md:671]: is this a complete sentence? seems missing noun :) Maybe reverse the order like you did in some other places. `Set the overwrite flag ... to overwrite data in the existing partitions.`?

File: docs/flink-getting-started.md [hunk 5]
```diff
@@ -710,7 +687,7 @@ env.execute("Test Iceberg DataStream");
 
 ### Upsert data
 
-To upsert the data in existing iceberg table, we could set the `upsert` flag in FlinkSink builder. The table must use v2 table format and have a primary key.
+To upsert the data in existing iceberg table, set the `upsert` flag in FlinkSink builder. The table must use v2 table format and have a primary key.
```
Comment (stevenzwu) [inline docs/flink-getting-started.md:690]: similarly reverse the order?

File: docs/flink-getting-started.md [hunk 6]
```diff
@@ -747,8 +724,8 @@ DataStream<org.apache.avro.generic.GenericRecord> dataStream = ...;
 
 Schema icebergSchema = table.schema();
 
-// if the Iceberg table schema contains time fields, we can't use
-// Avro schema converted from Iceberg schema via AvroSchemaUtil.
+// if the Iceberg table schema contains time fields, and can't use
+// the Avro schema converted from Iceberg schema via AvroSchemaUtil.
```
Comment (stevenzwu) [inline docs/flink-getting-started.md:728]: nit: seems odd syntax. maybe `the Avro schema converted from Iceberg schema can't be used due to precision difference between how Iceberg schema (micro) and Flink {@link AvroToRowDataConverters} (milli) deal with time type`

File: docs/flink-getting-started.md (hunk 1 above)
Comment (Fokko) [reply docs/flink-getting-started.md:49]: Oops, refactored that one too many times, thanks!

File: docs/flink-getting-started.md (hunk 3 above)
Comment (Fokko) [reply docs/flink-getting-started.md:327]: Good one, the url seems to be stable

File: docs/flink-getting-started.md (hunk 4 above)
Comment (Fokko) [reply docs/flink-getting-started.md:671]: Yes, the sentence doesn't read very easily. I like your suggestion very much. Updated!

**Review by stevenzwu — APPROVED** [review]

**Comment by stevenzwu** [comment]
thanks @Fokko for updating the doc

**Comment by Fokko** [comment]
Any time @stevenzwu thanks for the prompt review!

**Pull request merged**